src/Aquifer.Public.API/Endpoints/Bibles/Alignments/Greek/Endpoint.cs

```diff
@@ -0,0 +1,275 @@
+                    .QueryAsync<GreekSenseResult>(new CommandDefinition(GreekSensesQuery, new { greekSenseIds = batch }, cancellationToken: ct)));
+            }
+        }
+
+        var greekSenseResultById = greekSenseResults.ToLookup(r => r.Id);
+
+        var response = new Response
+        {
+            BibleId = bookData.BibleId,
+            BibleName = bookData.BibleName,
+            BibleAbbreviation = bookData.BibleAbbreviation,
+            GreekBibleAbbreviation = greekAlignmentSource,
+            BookCode = bookData.BookCode,
+            BookName = bookData.BookName,
+            Chapters = bibleText
+                .GroupBy(r => r.BibleWordIdentifier.Chapter)
+                .Select(chapterGrouping => new ResponseChapter
+                {
+                    Number = chapterGrouping.Key,
+                    Verses = chapterGrouping
+                        .GroupBy(r => r.BibleWordIdentifier.Verse)
+                        .Select(verseGrouping =>
+                        {
+                            var bibleVersionWordIdBookendsInBibleVersionWordGroupByBibleVersionWordGroupId = verseGrouping
+                                .GroupBy(r => r.BibleVersionWordGroupId)
+                                .ToDictionary(grp => grp.Key, grp => (First: grp.First().Id, Last: grp.Last().Id));
+
+                            return new ResponseChapterVerse
+                            {
+                                Number = verseGrouping.Key,
+                                Words = verseGrouping
+                                    .GroupBy(r => r.BibleWordIdentifier.Word)
+                                    .Select(wordGrouping =>
+                                    {
+                                        var word = wordGrouping.First();
+                                        return new EnglishWordWithGreekAlignment
+                                        {
+                                            Number = word.BibleWordIdentifier.Word,
+                                            Word = word.EnglishWord,
+                                            NextWordIsInGroup = bibleVersionWordIdBookendsInBibleVersionWordGroupByBibleVersionWordGroupId[word.BibleVersionWordGroupId].Last != word.Id,
+                                            GreekWords =
+                                                bibleVersionWordIdBookendsInBibleVersionWordGroupByBibleVersionWordGroupId[word.BibleVersionWordGroupId].First == word.Id
+                                                ? wordGrouping
+                                                    .GroupBy(r => r.GreekWordId)
+                                                    .Select(greekWordGrouping =>
+                                                    {
+                                                        var greekWordResult = greekWordResultById[greekWordGrouping.Key];
+                                                        var greekSenseResults = greekWordGrouping
```

Comment:
This local variable hides the outer greekSenseResults above, recommend changing the name of one of them.